mock-recipe-server/bin/upload.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+set -eu
+
+BUILD_DIR=$1
+S3_BUCKET=$2
+
+aws s3 rm --recursive "s3://$S3_BUCKET/"
+aws s3 cp --recursive "$BUILD_DIR" "s3://$S3_BUCKET/" --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers
```

Comment:
I think the way you have it is totally fine though. Just make sure `$S3_BUCKET` is set to avoid errant `rm`'s.